src/dmd/globals.d

```diff
@@ -34,6 +34,13 @@ enum TARGET : bool
     DragonFlyBSD = xversion!`DragonFlyBSD`,
 }
 
+enum Diagnostic : ubyte
+{
+    off,          // disable diagnostic
+    warn,         // generate a warning
```

Comment:
Please avoid abbreviations as well.

Reply:
1. If its not an error, it is a warning.

2. Having different default values was precisely the reason for two different types in the first place.

3. Warn is a word, not an abbreviation.

Reply:
1. It's a diagnostic message. Because if it's a warning, it should respect `-w`.

2. I don't see your point here ? They still have different default values, I'm just saying the first entry in the `enum` could be `error` to make using it slightly more robust (because if someone forgets to initialize it, it'll be obvious).

Reply:
1. `-w` produces an error.
2. Meh, the only two users explicitly sets the default value anyway.